%\VignetteIndexEntry{DEGseq}
%\VignetteKeywords{differentially expressed genes}
%\VignetteDepends{qvalue}
%\VignettePackage{DEGseq}

\documentclass[11pt]{article}
\usepackage{Sweave}
\usepackage{amsmath}
\usepackage{hyperref}
\usepackage[authoryear,round]{natbib}

\setlength{\textheight}{8.5in}
\setlength{\textwidth}{6in}
\setlength{\topmargin}{-0.25in}
\setlength{\oddsidemargin}{0.25in}
\setlength{\evensidemargin}{0.25in}
\newcommand{\Rpackage}[1]{{\textit{#1}}}

\begin{document}
\input{MeSH_db-concordance}
\title{\bf How to use the MeSH.db Package}
\author{Koki Tsuyuzaki$^1$, Itoshi Nikaido$^2$$^,$$^3$ and Gota Morota$^4$.}
\maketitle
\begin{center}
\noindent
$^1$Department of Medical and Life Science, Tokyo University of Science.\\
\noindent
$^2$Functional Genomics Unit, RIKEN Center for Developmental Biology.\\
\noindent
$^3$Laboratory for Systems Biology, RIKEN Center for Developmental Biology.\\
\noindent
$^4$Department of Dairy Science, University of Wisconsin-Madison.\\
\end{center}

\begin{center}
{\tt k.t.the-answer@hotmail.co.jp}
\end{center}
\tableofcontents

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Introduction}
This document provides the way to use \Rpackage{MeSH.db} package. MeSH (Medical Subject Headings) is the NLM controlled vocabulary used to manually index articles for MEDLINE/Pubmed
\citep{Nelson2004}. The amount of vocabulary in MeSH is about twice as large as that of GO (Gene Ontology)\citep{Ashburner2000} and its category is also wider. Therefore MeSH is expected to be much detailed and exhaustive annotation tool. Some softwares or databases using MeSH are now proposed \citep{Nakazato2007,Nakazato2009,Saurin2010,Sartor2012}.

\Rpackage{MeSH.db} is a free R package for handling MeSH in R. Its data are retrieved from MeSH ftp site (\url{http://www.nlm.nih.gov/mesh/filelist.html}). MeSH in 2013 has 19 hierarchy and MeSH.db provides 16 of them, which are actually assigned to some MeSH Terms. Each category is expressed to single capital alphabet defined in MeSH as Abbreviations.

\begin{center}
  \begin{table}[htbp]
    \begin{tabular}{|c|l|}\hline
      Abbreviation & Category \\ \hline \hline
      A & Anatomy \\ \hline
      B & Organisms \\ \hline
      C & Diseases \\ \hline
      D & Chemicals and Drugs \\ \hline
      E & Analytical, Diagnostic and Therapeutic Techniques and Equipment \\ \hline
      F & Psychiatry and Psychology \\ \hline
      G & Phenomena and Processes \\ \hline
      H & Disciplines and Occupations \\ \hline
      I & Anthropology, Education, Sociology and Social Phenomena \\ \hline
      J & Technology and Food and Beverages \\ \hline
      K & Humanities \\ \hline
      L & Information Science \\ \hline
      M & Persons \\ \hline
      N & Health Care \\ \hline
      V & Publication Type \\ \hline
      Z & Geographical Locations \\ \hline
\end{tabular}
  \end{table}
\end{center}
MeSH Term also have hierarchy structure like GO. \Rpackage{MeSH.db} provides its Ancestor-Offspring Relationships (AOR) and Parent-Child Relationships (PCR) as corresponding table. Data of PCR and AOR are also used for calculating the conditional probability in enrichment analysis (\Rpackage{meshr} package).
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Getting started}
To load the \Rpackage{MeSH.db} package, just type library(MeSH.db). 5 methods and 36 data are provided by \Rpackage{MeSH.db}.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Methods}
Following 5 methods are provided by \Rpackage{MeSH.db}.
\begin{center}
  \begin{table}[htbp]
    \begin{tabular*}{150mm}{@{\extracolsep{\fill}}|c|l|}\hline
      MeSH & Function for retrieval of the summary of all object in MeSH.db \\ \hline
      MeSH\_dbconn & Function for retrieval of the connection of .sqlite database\\ \hline
      MeSH\_dbfile & Function for retrieval of the directory of .sqlite file \\ \hline
      MeSH\_dbschema & Function for retrieval of the schema of .sqlite database \\ \hline
      MeSH\_dbInfo & Function for retrieval of the information of .sqlite database \\ \hline
\end{tabular*}
  \end{table}
\end{center}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Data}
Following 36 data are provided by \Rpackage{MeSH.db}.
\\\\
\begin{center}
  \begin{table}[htbp]
    \begin{tabular*}{150mm}{@{\extracolsep{\fill}}|p{40mm}|p{100mm}|}\hline
      MeSHMAPCOUNTS & The number of row of all data \\ \hline
      MeSHTERM & MeSH Term\\ \hline
      MeSHSYNONYM & The synonym of MeSH Term \\ \hline
      MeSHQUALIFIER & Substantial Information of MeSH Term \\ \hline \hline
      MeSHAAOR & Ancestor-Offspring Relationships in A category \\ \hline
      MeSHBAOR & Ancestor-Offspring Relationships in B category \\ \hline
      MeSHCAOR & Ancestor-Offspring Relationships in C category \\ \hline
      MeSHDAOR & Ancestor-Offspring Relationships in D category \\ \hline
      MeSHEAOR & Ancestor-Offspring Relationships in E category \\ \hline
      MeSHFAOR & Ancestor-Offspring Relationships in F category \\ \hline
      MeSHGAOR & Ancestor-Offspring Relationships in G category \\ \hline
      MeSHHAOR & Ancestor-Offspring Relationships in H category \\ \hline
      MeSHIAOR & Ancestor-Offspring Relationships in I category \\ \hline
      MeSHJAOR & Ancestor-Offspring Relationships in J category \\ \hline
      MeSHKAOR & Ancestor-Offspring Relationships in K category \\ \hline
      MeSHLAOR & Ancestor-Offspring Relationships in L category \\ \hline
      MeSHMAOR & Ancestor-Offspring Relationships in M category \\ \hline
      MeSHNAOR & Ancestor-Offspring Relationships in N category \\ \hline
      MeSHVAOR & Ancestor-Offspring Relationships in V category \\ \hline
      MeSHZAOR & Ancestor-Offspring Relationships in Z category \\ \hline \hline
      MeSHAPCR & Parent-Child Relationships in A category \\ \hline
      MeSHBPCR & Parent-Child Relationships in B category \\ \hline
      MeSHCPCR & Parent-Child Relationships in C category \\ \hline
      MeSHDPCR & Parent-Child Relationships in D category \\ \hline
      MeSHEPCR & Parent-Child Relationships in E category \\ \hline
      MeSHFPCR & Parent-Child Relationships in F category \\ \hline
      MeSHGPCR & Parent-Child Relationships in G category \\ \hline
      MeSHHPCR & Parent-Child Relationships in H category \\ \hline
      MeSHIPCR & Parent-Child Relationships in I category \\ \hline
      MeSHJPCR & Parent-Child Relationships in J category \\ \hline
      MeSHKPCR & Parent-Child Relationships in K category \\ \hline
      MeSHLPCR & Parent-Child Relationships in L category \\ \hline
      MeSHMPCR & Parent-Child Relationships in M category \\ \hline
    \end{tabular*}
  \end{table}
\end{center}

\begin{center}
  \begin{table}[htbp]
    \begin{tabular*}{150mm}{@{\extracolsep{\fill}}|p{40mm}|p{100mm}|}\hline
      MeSHNPCR & Parent-Child Relationships in N category \\ \hline
      MeSHVPCR & Parent-Child Relationships in V category \\ \hline
      MeSHZPCR & Parent-Child Relationships in Z category \\ \hline
    \end{tabular*}
  \end{table}
\end{center}
In \Rpackage{MeSH.db}, all data are extracted by 4 functions defined by \Rpackage{AnnotationForge}; $\bf{keytypes}$, $\bf{cols}$, $\bf{keys}$ and $\bf{select}$. keys function has 1 optional parameter keytype and select function also has 3 optional parameter keys, cols and keytype. cols is the columns which you can retrieved by select and keytype is the columns which you can specify as the option in keys and select functions.
\begin{center}
  \begin{table}[htbp]
    \begin{tabular*}{160mm}{@{\extracolsep{\fill}}|p{35mm}|p{55mm}|p{55mm}|} \hline
      Object Name & cols & keytype \\ \hline \hline
      MeSHMAPCOUNTS & MAPNAME, COUNT & MAPNAME \\ \hline
      MeSHTERM & \shortstack{MESHID, MESHTERM,\\ CATEGORY} & \shortstack{MESHID, MESHTERM,\\ CATEGORY} \\ \hline
      MeSHSYNONYM & MESHID, MESHSYNONYM & MESHID \\ \hline
      MeSHQUALIFIER & \shortstack{QUALIFIERID, SUBHEADING,\\ MESHID} & QUALIFIERID, MESHID \\ \hline \hline
      MeSHAAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHBAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHCAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHDAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHEAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHFAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHGAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHHAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHIAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHJAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHKAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHLAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
   \end{tabular*}
  \end{table}
\end{center}

\begin{center}
  \begin{table}[htbp]
    \begin{tabular*}{160mm}{@{\extracolsep{\fill}}|p{35mm}|p{55mm}|p{55mm}|} \hline
     MeSHMAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHNAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHVAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline
     MeSHZAOR & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} & \shortstack{ANCESTERMESHID,\\ OFFSPRINGMESHID} \\ \hline \hline
    MeSHAPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHBPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHCPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHDPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHEPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHFPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHGPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHHPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHIPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHJPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHKPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHLPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHMPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHNPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHVPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline
    MeSHZPCR & \shortstack{PARENTMESHID,\\ CHILDMESHID} & \shortstack{PARENTMESHID,\\ CHILDMESHID} \\ \hline     
   \end{tabular*}
  \end{table}
\end{center}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Examples}
\subsection{Exercises in cols, keytypes, keys and select method}
\Rpackage{MeSH.db uses} cols, keytypes, keys and select functions defined by \Rpackage{AnnotationForge}. Here we shows how to use these functions in \Rpackage{MeSH.db}.\\

First, install and load the \Rpackage{MeSH.db}.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> library(MeSH.db)
\end{Sinput}
\end{Schunk}
\end{center}


ls shows all object in \Rpackage{MeSH.db}.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> ls("package:MeSH.db")
\end{Sinput}
\begin{Soutput}
 [1] "MeSH"          "MeSH_dbconn"   "MeSH_dbfile"   "MeSH_dbInfo"  
 [5] "MeSH_dbschema" "MeSHAAOR"      "MeSHAPCR"      "MeSHBAOR"     
 [9] "MeSHBPCR"      "MeSHCAOR"      "MeSHCPCR"      "MeSHDAOR"     
[13] "MeSHDPCR"      "MeSHEAOR"      "MeSHEPCR"      "MeSHFAOR"     
[17] "MeSHFPCR"      "MeSHGAOR"      "MeSHGPCR"      "MeSHHAOR"     
[21] "MeSHHPCR"      "MeSHIAOR"      "MeSHIPCR"      "MeSHJAOR"     
[25] "MeSHJPCR"      "MeSHKAOR"      "MeSHKPCR"      "MeSHLAOR"     
[29] "MeSHLPCR"      "MeSHMAOR"      "MeSHMAPCOUNTS" "MeSHMPCR"     
[33] "MeSHNAOR"      "MeSHNPCR"      "MeSHQUALIFIER" "MeSHSYNONYM"  
[37] "MeSHTERM"      "MeSHVAOR"      "MeSHVPCR"      "MeSHZAOR"     
[41] "MeSHZPCR"     
\end{Soutput}
\end{Schunk}
\end{center}


Here we use cols, keytypes, keys and select against MeSHMAPCOUNTS.\\

cols returns the rows which you can retrieve in MeSHMAPCOUNTS.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> cols(MeSHMAPCOUNTS)
\end{Sinput}
\begin{Soutput}
[1] "MAPNAME" "COUNT"  
\end{Soutput}
\end{Schunk}
\end{center}


keytypes returns the rows which you can use the option as keys and select functions against MeSHMAPCOUNTS.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> keytypes(MeSHMAPCOUNTS)
\end{Sinput}
\begin{Soutput}
[1] "MAPNAME"
\end{Soutput}
\end{Schunk}
\end{center}
Here we get to know that MAPNAME is available.

\newpage
keys function specifies the value of keytype.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> k <- keys(MeSHMAPCOUNTS, keytype="MAPNAME")
> head(k)
\end{Sinput}
\begin{Soutput}
        MAPNAME
1      MeSHTERM
2   MeSHSYNONYM
3 MeSHQUALIFIER
4      MeSHAAOR
5      MeSHBAOR
6      MeSHCAOR
\end{Soutput}
\end{Schunk}
\end{center}


select method specifies the rows in particular cols having user-defined keys and retrieved data as single dataframe like SQL's SELECT statement. Now we retrieve the rows in which MAPNAME is equivalent to "MeSHTERM".
\begin{center}
\begin{Schunk}
\begin{Sinput}
> select(MeSHMAPCOUNTS, keys=k[1,], cols=c("MAPNAME","COUNT"),
+        keytype="MAPNAME")
\end{Sinput}
\begin{Soutput}
   MAPNAME COUNT
1 MeSHTERM 54843
\end{Soutput}
\end{Schunk}
\end{center}


By the way, here we don't have to specify keytype option against MeSHMAPCOUNTS, because MeSHMAPCOUNTS only has single col which is possible to be keytype and keytype is consequently specified.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> select(MeSHMAPCOUNTS, keys=k[1,], cols=c("MAPNAME","COUNT"))
\end{Sinput}
\begin{Soutput}
   MAPNAME COUNT
1 MeSHTERM 54843
\end{Soutput}
\end{Schunk}
\end{center}
The same can be said of MeSHSYNONYM.

\newpage
\subsection{Annotation of $Cystic$ $Fibrosis$}


Next we will annotate one of genetic diseases, $Cystic$ $Fibrosis$ ($CF$) by MeSH. Let's try to search $CF$ in MeSHTERM.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> cols(MeSHTERM)
\end{Sinput}
\begin{Soutput}
[1] "MESHID"       "MESHTERM"     "MESHCATEGORY"
\end{Soutput}
\end{Schunk}
\end{center}


MESHID, MESHTERM and MESHCATEGORY can be retrieved and ...
\begin{center}
\begin{Schunk}
\begin{Sinput}
> keytypes(MeSHTERM)
\end{Sinput}
\begin{Soutput}
[1] "MESHID"       "MESHTERM"     "MESHCATEGORY"
\end{Soutput}
\end{Schunk}
\end{center}
all of them are available as keytype option.\\


select function retrieves the rows in which MESHTERM is "$Cystic$ $Fibrosis$" in MeSHTERM table.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> CF <- select(MeSHTERM, keys="Cystic Fibrosis",
+             cols=c("MESHID", "MESHTERM", "MESHCATEGORY"), keytype="MESHTERM")
> CF
\end{Sinput}
\begin{Soutput}
   MESHID        MESHTERM MESHCATEGORY
1 D003550 Cystic Fibrosis            C
\end{Soutput}
\end{Schunk}
\end{center}


select function shows that MESHID of $CF$ is D003550 and $CF$ is in C (Disease) category.\\


Using MeSHSYNONYM, we can also check whether $CF$ has synonyms.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> select(MeSHSYNONYM, keys=CF[1,1],
+        cols=c("MESHID","MESHSYNONYM"), keytype="MESHTERM")
\end{Sinput}
\end{Schunk}
\end{center}
\begin{center}
\begin{Schunk}
\begin{Soutput}
MESHID
1 D003550
3 D003550
4 D003550
5 D003550
6 D003550
7 D003550
8 D003550
9 D003550
10 D003550
MESHSYNONYM
1 Mucoviscidosis|T047|NON|EQV|OMIM (2013)|ORD (2010)|UNK
(19XX)|740329|abcdeeef
3 Fibrocystic Disease of Pancreas|T047|NON|NRW|UNK
(19XX)|740329|FIBROCYSTIC DIS OF PANCREAS|abcdefv
4 Pancreatic Cystic Fibrosis|T047|NON|NRW|UNK (19XX)|740329|abcdef
5 Pulmonary Cystic Fibrosis|T047|NON|NRW|NLM (2005)|031030|PULM CYSTIC
FIBROSIS|abcdefv
6 Cystic Fibrosis, Pancreatic
7 Cystic Fibrosis, Pulmonary
8 Fibrosis, Cystic
9 Pancreas Fibrocystic Disease
10 Pancreas Fibrocystic Diseases
\end{Soutput}
\end{Schunk}
\end{center}
We can get to know that $CF$ has some synonyms like Mucoviscidosis, Fibrocystic Disease of Pancreas, Pancreastic Cystic Fibrosis and so on.\\\\


MeSH also defines QUALIFIER, which is more rough category (SUBHEADING). We can also use select function against MeSHQUALIFIER.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> select(MeSHQUALIFIER, keys=CF[1,1], 
+        cols=c("QUALIFIERID","SUBHEADING","MESHID"), keytype="MESHID")
\end{Sinput}
\begin{Soutput}
   QUALIFIERID           SUBHEADING  MESHID
1      Q000097                blood D003550
2      Q000134  cerebrospinal fluid D003550
3      Q000139   chemically induced D003550
4      Q000145       classification D003550
5      Q000150        complications D003550
6      Q000175            diagnosis D003550
7      Q000178         diet therapy D003550
8      Q000188         drug therapy D003550
9      Q000191            economics D003550
10     Q000196           embryology D003550
11     Q000201           enzymology D003550
12     Q000208            ethnology D003550
13     Q000209             etiology D003550
14     Q000235             genetics D003550
15     Q000266              history D003550
16     Q000276           immunology D003550
17     Q000378           metabolism D003550
18     Q000382         microbiology D003550
19     Q000401            mortality D003550
20     Q000451              nursing D003550
21     Q000453         epidemiology D003550
22     Q000469         parasitology D003550
23     Q000473            pathology D003550
24     Q000503      physiopathology D003550
25     Q000517 prevention & control D003550
26     Q000523           psychology D003550
27     Q000530          radiography D003550
28     Q000531 radionuclide imaging D003550
29     Q000532         radiotherapy D003550
30     Q000534       rehabilitation D003550
31     Q000601              surgery D003550
32     Q000628              therapy D003550
33     Q000652                urine D003550
34     Q000662           veterinary D003550
35     Q000736      ultrasonography D003550
36     Q000821             virology D003550
\end{Soutput}
\end{Schunk}
\end{center}


As mentioned before, MeSH has hierarchical structure. AOR tell us upper (or lower) hierarchical MeSH Term. We already know $CF$ is categorized in C, so MeSH$\bf{C}$AOR is suitable here.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> ao <- select(MeSHCAOR, keys=CF[1,1], 
+        cols=c("ANCESTORMESHID","OFFSPRINGMESHID"), keytype="OFFSPRINGMESHID")
> ao
\end{Sinput}
\begin{Soutput}
  ANCESTORMESHID OFFSPRINGMESHID
1        D007232         D003550
2        D008171         D003550
3        D010182         D003550
4        D030342         D003550
\end{Soutput}
\end{Schunk}
\end{center}
There are D007232, D008171, D010182 and D030342 above $CF$.\\


We will translate these MeSH ID to MeSH Term. 
\begin{center}
\begin{Schunk}
\begin{Sinput}
> select(MeSHTERM, keys=ao[,1], cols=c("MESHTERM"), keytype="MESHID")
\end{Sinput}
\begin{Soutput}
                   MESHTERM
1 Infant, Newborn, Diseases
2             Lung Diseases
3       Pancreatic Diseases
4  Genetic Diseases, Inborn
\end{Soutput}
\end{Schunk}
\end{center}
$CF$ is recognized as various kind of diseases.


PCR tell us the direct upper (or lower) MeSH Term. select function is applied to MeSH$\bf{C}$PCR in the same way.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> pc <- select(MeSHCPCR, keys=CF[1,1],
+        cols=c("PARENTMESHID","CHILDMESHID"), keytype="CHILDMESHID")
> pc
\end{Sinput}
\begin{Soutput}
  PARENTMESHID CHILDMESHID
1      D010182     D003550
2      D008171     D003550
3      D030342     D003550
4      D007232     D003550
\end{Soutput}
\end{Schunk}
\end{center}
Same MeSH IDs are retrieved, which means MeSH Term of $CF$ is assigned in different branches of MeSH hierarchy separetelly.\\



Once we set keytype to opposite direction (OFFSPRINGMESHID to ANCESTORMESHID), we can also retrieved MeSH ID in lower hierarchy.
\begin{center}
\begin{Schunk}
\begin{Sinput}
> select(MeSHCAOR, keys=CF[1,1], 
+        cols=c("ANCESTORMESHID","OFFSPRINGMESHID"), keytype="ANCESTORMESHID")
\end{Sinput}
\begin{Soutput}
[1] ANCESTORMESHID  OFFSPRINGMESHID
<0 rows> (or 0-length row.names)
\end{Soutput}
\end{Schunk}
\end{center}
There are any MeSH ID, which means $CF$ has no lower hierarchy.\\\\


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\newpage
\bibliographystyle{apalike}
\bibliography{MeSH_db}
\end{document}
